lib/resty/aes.lua

```diff
@@ -126,8 +105,19 @@ end
 _M.cipher = cipher
 
 function _M.new(self, key, salt, _cipher, _hash, hash_rounds)
-    local encrypt_ctx = ffi_new(ctx_ptr_type)
-    local decrypt_ctx = ffi_new(ctx_ptr_type)
+    local encrypt_ctx = C.EVP_CIPHER_CTX_new()
+    if encrypt_ctx == nil then
+        return nil, "no memory"
+    end
+
+    local decrypt_ctx = C.EVP_CIPHER_CTX_new()
+    if decrypt_ctx == nil then
+        return nil, "no memory"
+    end
+
+    ffi_gc(encrypt_ctx, C.EVP_CIPHER_CTX_free)
```

Comment:
We should put this line earlier otherwise there might be a memory leak here. (I can see the original code does have this risk)